homeassistant/components/synology_dsm/diagnostics.py

```diff
@@ -0,0 +1,101 @@
+"""Diagnostics support for Synology DSM."""
+from __future__ import annotations
+
+from synology_dsm.api.surveillance_station.camera import SynoCamera
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import SynoApi
+from .const import DOMAIN, SYNO_API, SYSTEM_LOADED
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict:
+    """Return diagnostics for a config entry."""
+    data: dict = hass.data[DOMAIN][entry.unique_id]
+    syno_api: SynoApi = data[SYNO_API]
+    dsm_info = syno_api.dsm.information
+
+    diag_data = {
+        "entry": {
```

Comment:
you can also use `entry.as_dict()`

Reply:
but make sure you redact any potential tokens in entry.data using the redact data helper